dali/operators/decoder/video/video_decoder.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/decoder/video/video_decoder.h"
+
+namespace dali {
+
+template <>
+bool VideoDecoder<CPUBackend>::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                         const workspace_t<CPUBackend> &ws) {
+  ValidateInput(ws);
+  const auto &input = ws.Input<CPUBackend>(0);
+  int batch_size = input.num_samples();
+  frames_decoders_.resize(batch_size);
+  auto &thread_pool = ws.GetThreadPool();
+  for (int i = 0; i < batch_size; ++i) {
+    auto sample = input[i];
+    auto data = reinterpret_cast<const char *>(sample.data<uint8_t>());
+    size_t size = sample.shape().num_elements();
+    thread_pool.AddWork([this, i, data, size](int tid) {
+      frames_decoders_[i] = std::make_unique<FramesDecoder>(data, size);
+    }, volume(sample.shape()));
```

Comment:
Do we need to set any priority here? I guess the execution time is the same for all videos (it mostly creation of the context and parsing of the header).

Reply:
Currently, this also builds index, so I guess the amount of work might depend on number of frames

Reply:
I think we can remove that pretty soon.